ml-agents/mlagents/torch_utils/cpu_utils.py

```diff
@@ -20,8 +20,13 @@ def _get_num_available_cpus() -> Optional[int]:
     """
     period = _read_in_integer_file("/sys/fs/cgroup/cpu/cpu.cfs_period_us")
     quota = _read_in_integer_file("/sys/fs/cgroup/cpu/cpu.cfs_quota_us")
+    share = _read_in_integer_file("/sys/fs/cgroup/cpu/cpu.share")
     if period > 0 and quota > 0:
```

Comment:
So I was wrong about `cpu.shares` not existing on a raw Ubuntu VM. On a 4-core bare-metal machine, I have:

- `cpu.shares` = 1024
- `cfs_period_us` = 100000
- `cfs_quota_us` = -1

So Torch would only see 1 core with this logic. Argh.

Reply:
I added logic so this code only runs if on kubernetes.